doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,51 @@
+
+## Troubleshooting
+
+### Common Pitfalls
+#### Login profile files
+When the SSH slaves plugin connects to a slave, it does not run an interactive shell. 
+Instead it does the equivalent of your running `ssh slavehost command...` a few times, 
+eventually to run `ssh slavehost java -jar ...`. Exactly what happens on the slave as a result of this depends on the SSHD implementation, 
+but OpenSSH runs this with `bash -c command ...` (or whatever your login shell is.)
```

Comment:
I'd suggest removing the "but" and separating this into two sentences -> "SSHD implementation. OpenSSH runs"